Generator/Templates/Shared/helper_dll_import.sbntxt

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace {{ namespace }}
+{
+    internal static class DllImport
+    {
+        #region Fields
+        
+        private const string _windowsDllName = "{{ windows_dll }}";
+        private const string _linuxDllName = "{{ linux_dll }}";
+        private const string _osxDllName = "{{ osx_dll }}";
+
+        private static readonly string DllName;
+
+        #endregion
+        
+        #region Constructors
+        
+        static DllImport()
+        {
+            DllName = GetDllImport();
+        }
+        
+        #endregion
+            
+        #region Methods
+        
+        public static void Initialize()
+        {
+            NativeLibrary.SetDllImportResolver(typeof(DllImport).Assembly, ImportResolver);
+        }
+        
+        private static IntPtr ImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            if (NativeLibrary.TryLoad(DllName, assembly, DllImportSearchPath.System32, out IntPtr libHandle))
+                return libHandle;
```

Comment:
Maybe a way to cache the loaded library, to avoid redundant loading ?

Reply:
I'm not sure if there is some caching on the side of `NativeLibrary.TryLoad`.

This reads a little bit like it: [docs](https://docs.microsoft.com/en-us/dotnet/core/dependency-loading/loading-unmanaged).

If you agreee I would avoid adding caching for now. If we see problems later or have definitive feedback that we are missing an opportunity here we can add the caching ourself.

@gircore/coreteam what do you think?

Reply:
Hm I added a `Console.WriteLine` and the `ImportResolver` method is called several times. So it looks like it is not caching. But I'm not sure if we should add caching and if this could result in some unexpected results.

Reply:
I think we can check that by printing the returned IntPtr to the console, if the result is the same may be it's finally better to cache ?
But I've no problem to skip this for now.